secrets/azurekeyvault/akv.go

```diff
@@ -254,6 +254,9 @@ func (k *keeper) getKeyVaultURI() string {
 }
 
 func (k *keeper) validateOptions() error {
+	if k.options == nil {
```

Comment:
Instead of checking this separately, how about we replace `nil` Options with `&Options{}` in the constructor? The check for empty algorithm will still fail, but that's a better error message.